src/SFA.DAS.ApplyService.Web/Services/OverallOutcomeAugmentationService.cs

```diff
@@ -0,0 +1,334 @@
+
+            foreach (var sequence in sequencesWithModerationFails)
+            {
+                var sectionsToRemove = new List<AssessorSection>();
+                foreach (var section in sequence.Sections)
+                {
+                    var pagesToRemove = new List<Page>();
+
+                    foreach (var page in section.Pages)
+                    {
+                        var selectedSection = allSections.FirstOrDefault(x =>
+                            x.SequenceId == section.SequenceNumber && x.SectionId == section.SectionNumber);
+
+                        var pageDetails =
+                            selectedSection?.QnAData?.Pages?.SingleOrDefault(p => p.PageId == page.PageId);
+                        if (pageDetails == null || !pageDetails.Active)
+                        {
+                            pagesToRemove.Add(page);
+                        }
+                    }
+
+                    if (pagesToRemove.Any())
+                    {
+                        foreach (var pageToRemove in pagesToRemove)
+                            section.Pages.Remove(pageToRemove);
+                    }
+
+                    if (section.Pages.Count==0)
+                        sectionsToRemove.Add(section);
+                }
+
+                if (sectionsToRemove.Any())
+                {
+                    foreach (var sectionToRemove in sectionsToRemove)
+                    {
+                        sequence.Sections.Remove(sectionToRemove);
+                    }
+                }
+
+                if (sequence.Sections.Count == 0)
+                {
+                    sequencesToRemove.Add(sequence);
+                }
+            }
+
+            if (sequencesToRemove.Any())
+            {
+                foreach (var sequence in sequencesToRemove)
+                {
+                    sequencesWithModerationFails.Remove(sequence);
+                }
+            }
+        }
+
+        private static void BuildSequencesWithModerationFails(List<AssessorSequence> sequences,
+            List<AssessorSequence> sequencesWithModerationFails)
+        {
+            foreach (var sequence in sequences)
+            {
+                foreach (var section in sequence.Sections)
+                {
+                    if (section.Pages != null)
+                    {
+                        if (sequencesWithModerationFails.All(x => x.SequenceNumber != sequence.SequenceNumber))
+                            sequencesWithModerationFails.Add(new AssessorSequence
+                            {
+                                SequenceNumber = sequence.SequenceNumber,
+                                SequenceTitle = sequence.SequenceTitle,
+                                Sections = new List<AssessorSection>()
+                            });
+
+                        sequencesWithModerationFails.FirstOrDefault(x => x.SequenceNumber == section.SequenceNumber)
+                            ?.Sections.Add(section);
```

Comment:
I'd probably expand this out to make it more readable as it conveniently hides the null check and add